tools/benchmarks/run_service_api.ipynb

```diff
@@ -0,0 +1,184 @@
+    "        # If the pipeline you choose needs to specify parameters to create a run, specify it here.\n",
+    "        parameters = []\n",
+    "        parameter = kfp_server_api.ApiParameter(name='pipeline-root', value='gs://jingzhangjz-project-outputs/tfx_taxi_simple/{{workflow.uid}}')\n",
+    "        parameters.append(parameter)\n",
+    "        parameter = kfp_server_api.ApiParameter(name='data-root', value='gs://ml-pipeline-playground/tfx_taxi_simple/data')\n",
+    "        parameters.append(parameter)\n",
+    "        parameter = kfp_server_api.ApiParameter(name='module-file', value='gs://ml-pipeline-playground/tfx_taxi_simple/modules/taxi_utils.py')\n",
+    "        parameters.append(parameter)        \n",
+    "        pipeline_spec = kfp_server_api.ApiPipelineSpec(parameters=parameters)\n",
+    "\n",
+    "        start = time.perf_counter()\n",
+    "        run_name = 'run-' + random_suffix()\n",
+    "        run = client.runs.create_run(body={'name':run_name, 'resource_references': resource_references, 'pipeline_spec': pipeline_spec})  \n",
+    "        dur = time.perf_counter() - start\n",
+    "        create_run_latencies.append(dur)   \n",
+    "        created_runs.append(run.run.id)\n",
+    "               \n",
+    "    # Wait for the runs to finish.\n",
+    "    while True:\n",
+    "        all_finished = True\n",
+    "        for i in created_runs:\n",
+    "            run = client.runs.get_run(i)  \n",
+    "            if not run_finished(run.run.status):\n",
+    "                all_finished = False\n",
+    "                break\n",
+    "        if all_finished:     \n",
+    "            break\n",
+    "        else:\n",
+    "            time.sleep(run_status_polling_interval_sec)\n",
+    "\n",
+    "    # When all runs are finished, measure run durations and get run latencies.\n",
+    "    get_run_latencies = []\n",
+    "    run_durations = []\n",
+    "    for i in created_runs:\n",
+    "        start = time.perf_counter()\n",
+    "        run = client.runs.get_run(i)  \n",
+    "        dur = time.perf_counter() - start\n",
+    "        get_run_latencies.append(dur) \n",
+    "        run_durations.append((run.run.finished_at - run.run.created_at).total_seconds())\n",
+    "\n",
+    "    # Measure delete run latency.\n",
+    "    delete_run_latencies = []\n",
+    "    for i in created_runs:\n",
+    "        start = time.perf_counter()\n",
+    "        run = client.runs.delete_run(i)  \n",
+    "        dur = time.perf_counter() - start\n",
+    "        delete_run_latencies.append(dur)   \n",
+    "    \n",
+    "    # Cleanup\n",
+    "    client.pipelines.delete_pipeline(pipeline.id)\n",
+    "    client.experiments.delete_experiment(experiment.id)\n",
+    "         \n",
+    "    # Plots\n",
+    "    fig, axs = plt.subplots(nrows=4, figsize=(10,20))\n",
+    "    \n",
+    "    label_create_run_latencies = pd.Series(create_run_latencies, name='Create Run Latency (Second)')\n",
```

Comment:
Another important plot I'd like to see is how many runs suceeded/failed

Reply:
done